modules/create-a-kubeletconfig-crd-to-edit-kubelet-parameters.adoc

```diff
@@ -16,12 +16,32 @@ As the fields in the `kubeletConfig` object are passed directly to the kubelet f
 
 Consider the following guidance:
 
-* Create one `KubeletConfig` CR for each machine config pool with all the config changes you want for that pool. If you are applying the same content to all of the pools, you need only one `KubeletConfig` CR for all of the pools.
-
 * Edit an existing `KubeletConfig` CR to modify existing settings or add new settings, instead of creating a CR for each change. It is recommended that you create a CR only to modify a different machine config pool, or for changes that are intended to be temporary, so that you can revert the changes.
 
+* Create one `KubeletConfig` CR for each machine config pool with all the config changes you want for that pool. If you are applying the same content to all of the pools, you need only one `KubeletConfig` CR for all of the pools.
+
 * As needed, create multiple `KubeletConfig` CRs with a limit of 10 per cluster. For the first `KubeletConfig` CR, the Machine Config Operator (MCO) creates a machine config appended with `kubelet`. With each subsequent CR, the controller creates another `kubelet` machine config with a numeric suffix. For example, if you have a `kubelet` machine config with a `-2` suffix, the next `kubelet` machine config is appended with `-3`.
 
+[NOTE]
+====
+If you are applying a kubelet or container runtime config to a custom machine config pool, the custom role in the `machineConfigSelector` must match the name of the custom machine config pool. 
```

Comment:
Do you think we should elaborate on why this is the case? I guess my concern is that this is how the MCP is structured, but doesn't really explain how to set up the kubeletconfig/containerruntimeconfig itself, so it might confuse readers.

the content itself is correct

Reply:
@yuqi-zhang If you have some context that will help the user understand this limitation, I would be happy to put it in.

Reply:
Sorry, I missed your comment. I was thinking of something like:

This is generally not the case, but for kubelet and containerruntime config objects, the generated machineconfig that lays down the actual configuration always references the role as the generated machineconfig role. Thus if they do not match, the MCO will still generate the machineconfig but the MCP will not match it by default.

I'm not sure if that's too verbose though and could be even more confusing